R/classicalpredictionperformance.R

```diff
@@ -0,0 +1,633 @@
+        t       = fatFits[[i]]$tval,
+        df      = fatFits[[i]]$df,
+        p       = fatFits[[i]]$pval
+      ))
+    }
+
+  }
+
+
+  return()
+}
+.metamiscFitFunnelAsymmetryPlot  <- function(jaspResults, options) {
+
+  if (!is.null(jaspResults[["funnelTestPlots"]])) {
+    funnelTestPlots <- jaspResults[["funnelTestPlots"]]
+  } else{
+    funnelTestPlots <- createJaspContainer(title = gettext("Funnel Plot Asymmetry Plots"))
+    funnelTestPlots$dependOn(c(.metamiscDependencies, "funnelAsymmetryTestPlot", "funnelAsymmetryTest", if (options[["method"]] == "BAYES") .metamiscDependenciesBayesian))
+    funnelTestPlots$position <- 4
+    jaspResults[["funnelTestPlots"]] <- funnelTestPlots
+  }
+
+  fatFits <- jaspResults[["modelsFat"]]$object
+
+  for(i in seq_along(fatFits)) {
+    if (!any(class(fatFits[[i]]) %in% c("simpleError", "error")) && is.null(funnelTestPlots[[.metamiscFitFunnelAsymmetryNames(fatFits[[i]]$method)]])) {
+
+      tempFunnelPlot   <- createJaspPlot(
+        title  = .metamiscFitFunnelAsymmetryNames(fatFits[[i]]$method),
+        width  = 340,
+        height = 300)
+      tempFunnelPlot$position <- i
+      tempFunnelPlot$dependOn(.metamiscFitFunnelAsymmetryOptions(fatFits[[i]]$method))
+      funnelTestPlots[[.metamiscFitFunnelAsymmetryNames(fatFits[[i]]$method)]] <- tempFunnelPlot
+
+      tempPlot <- tryCatch(.metamiscFitFunnelAsymmetryggPlot(fatFits[[i]]), error = function(e)e)
+
+      if (any(class(tempPlot) %in% c("simpleError", "error"))) {
+        tempFunnelPlot$setError(tempPlot$message)
+      } else{
+        tempPlot <- jaspGraphs::themeJasp(tempPlot)
+        tempFunnelPlot$plotObject <- tempPlot
+      }
+
+    }
+  }
+
+  return()
+}
+.metamiscFitFunnelAsymmetryNames <- function(shortcut) {
+  switch(
+    shortcut,
+    "E-UW"  = "Egger (unweighted)",
+    "E-FIV" = "Egger (multiplicative overdispersion)",
+    "M-FIV" = "Macaskill",
+    "M-FPV" = "Macaskill (pooled)",
+    "P-FPV" = "Peters",
+    "D-FIV" = "Debray"
+  )
+}
+.metamiscFitFunnelAsymmetryOptions <- function(shortcut) {
+  switch(
+    shortcut,
+    "E-UW"  = "funnelAsymmetryTestEggerUW",
+    "E-FIV" = "funnelAsymmetryTestEggerFIV",
+    "M-FIV" = "funnelAsymmetryTestMacaskillFIV",
+    "M-FPV" = "funnelAsymmetryTestMacaskillFPV",
+    "P-FPV" = "funnelAsymmetryTestPeters",
+    "D-FIV" = "funnelAsymmetryTestDebrayFIV"
+  )
+}
+.metamiscFitFunnelAsymmetryggPlot  <- function(x, ref, xlab = "Effect size",
+                                               confint = TRUE, confint.level = 0.1, confint.alpha = .50, confint.col = "skyblue") {
+
+  if (!inherits(x, "fat"))
+    stop("Argument 'x' must be an object of class \"fat\".")
+  if (confint.level < 0 | confint.level > 1) {
+    stop("Argument 'confint.level' must be between 0 and 1.")
+  }
+
+  xval <- x$model$data[, "y"]
+  if (x$method %in% c("E-UW", "E-FIV")) {
+    ylab <- "Standard error"
+    yval <- (x$model$data[, "x"])
+    ylim <- rev(c(0, max(yval, na.rm = T)))
```

Comment:
`T` -> `TRUE`.

Reply:
yeah, I just copied the code from the package to update the base plot functions to ggplot. Should be eventually taken care of withing the package itself